src/GdbCommandHandler.cc

```diff
@@ -42,11 +42,16 @@ python
 import re
 
 def gdb_unescape(string):
+    len = len(string)
     result = ""
-    pos = 0
-    while pos < len(string):
-        result += chr(int(string[pos:pos+2], 16))
-        pos += 2
+    try:
+        pos = 0
+        while pos < len:
+            hex_char = string[pos:pos+2]
+            result += chr(int(hex_char, 16))
+            pos += 2
+    except: # care for unexpected string length and value
+        return ""
```

Comment:
`string[pos:pos+2]` does not throw an exception if `pos == len(string) - 1`. So you probably should check `pos + 2 <= len`.